### DISCUSSION TITLE
Fastlane Build Successful but app crashes on launch

### DISCUSSION THREAD

**Discussion by umrashrf** [fastlane/fastlane#20284] [Q&A]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I have successfully released an iPhone app built by Fastlane on CircleCI but after downloading the app from TestFlight, the app crashes on launch. I connected Xcode and from Device and Simulators, I can see logs and stack trace (attached).

Trace: https://gist.github.com/umrashrf/5c1e65fd75b06bb1d832106a1bb20418

##### Command executed

https://gist.github.com/umrashrf/512fb449c5a3c6726e74d01a73ab7e58

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> https://gist.github.com/umrashrf/512fb449c5a3c6726e74d01a73ab7e58 </pre>
</details>

### Environment

<details>
  <pre> [⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [⠧] 🚀 [⠇] 🚀 [⠏] 🚀 [⠋] 🚀 [⠙] 🚀 [✔] 🚀 
[02:23:13]: fastlane detected a Gemfile in the current directory
[02:23:13]: However, it seems like you didn't use `bundle exec`
[02:23:13]: To launch fastlane faster, please use
[02:23:13]: 
[02:23:13]: $ bundle exec fastlane env --verbose
[02:23:13]: 
[02:23:13]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
Successfully loaded Appfile at path '/Users/distiller/project/fastlane/Appfile'
- app_identifier: 'com.website.AppName'
- apple_id: 'appleid@gmail.com'
- itc_team_id: '123456789'
- team_id: 'AB1C2345DD'
- json_key_file: 'play-store-credentials.json'
- package_name: 'com.website.AppName'
-------
INFO [2022-05-16 02:23:14.63]: Sending anonymous analytics information
INFO [2022-05-16 02:23:14.63]: Learn more at https://docs.fastlane.tools/#metrics
INFO [2022-05-16 02:23:14.63]: No personal or sensitive data is sent.
INFO [2022-05-16 02:23:14.63]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
WARN [2022-05-16 02:23:14.67]: Generating fastlane environment output, this might take a few seconds...
swift-driver version: 1.45.2 DEBUG [2022-05-16 02:23:16.60]: Checking if there are any plugins that should be loaded...
DEBUG [2022-05-16 02:23:16.66]: Loading 'fastlane-plugin-increment_version_code' plugin
WARN [2022-05-16 02:23:16.85]: Installing Ruby gem 'fastlane-plugin-increment_version_code'...
Fetching fastlane-plugin-increment_version_code-0.4.3.gem
INFO [2022-05-16 02:23:17.73]: Successfully installed 'fastlane-plugin-increment_version_code'
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                              |
| --------------------------- | -------------------------------------------------- |
| OS                          | 12.3.1                                             |
| Ruby                        | 2.7.5                                              |
| Bundler?                    | false                                              |
| Git                         | git version 2.35.1                                 |
| Installation Source         | ~/.gem/ruby/2.7.5/bin/fastlane                     |
| Host                        | macOS 12.3.1 (21E258)                              |
| Ruby Lib Dir                | ~/.rubies/ruby-2.7.5/lib                           |
| OpenSSL Version             | OpenSSL 1.0.2t  10 Sep 2019                        |
| Is contained                | false                                              |
| Is homebrew                 | false                                              |
| Is installed via Fabric.app | false                                              |
| Xcode Path                  | /Applications/Xcode-13.3.1.app/Contents/Developer/ |
| Xcode Version               | 13.3.1                                             |
| Swift Version               | 5.6                                                |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)

platform :ios do
  before_all do |lane|
    ENV["MATCH_KEYCHAIN_NAME"] = "circle-ci"
    ENV["MATCH_KEYCHAIN_PASSWORD"] = "*********"
    create_keychain(
      name: ENV["MATCH_KEYCHAIN_NAME"],
      password: ENV["MATCH_KEYCHAIN_PASSWORD"],
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      add_to_search_list: true
    )
    import_certificate(
      certificate_path: "AppleWWDRCAG3.cer",
      keychain_name: ENV["MATCH_KEYCHAIN_NAME"],
      keychain_password: ENV["MATCH_KEYCHAIN_PASSWORD"]
    )
    match(
      type: "appstore", 
      app_identifier: "com.website.AppName", 
      readonly: true,
      keychain_name: ENV["MATCH_KEYCHAIN_NAME"],
      keychain_password: ENV["MATCH_KEYCHAIN_PASSWORD"]
    )
  end

  after_all do |lane, options|
    delete_keychain(name: ENV["MATCH_KEYCHAIN_NAME"])
  end  

  desc 'Build the iOS application and push to Testflight.'
  lane :beta do
    api_key = app_store_connect_api_key(
      key_id: ENV["APP_STORE_CONNECT_API_KEY_KEY_ID"],
      issuer_id: ENV["APP_STORE_CONNECT_API_KEY_ISSUER_ID"],
      key_content: ENV["APP_STORE_CONNECT_API_KEY_KEY"],
      duration: 1200,
      in_house: false
    )
    increment_build_number(xcodeproj: "./ios/RentCar.xcodeproj")
    build_app(
      workspace: "./ios/RentCar.xcworkspace", 
      scheme: "RentCar",
      export_method: "app-store",
      export_options: {
        method: "app-store",
        provisioningProfiles: {
          "com.website.AppName" => "match AppStore com.website.AppName",
        }
      }
    )
    upload_to_testflight(api_key: api_key, skip_waiting_for_build_processing: true)
  end

  lane :bump do
    paths = [
      "ios/RentCar.xcodeproj/project.pbxproj",
      "ios/RentCar/Info.plist"
    ]
    git_add(path: paths)
    git_commit(path: paths, message: "iOS Version Bump")
    push_to_git_remote()
  end
end

platform :android do
  desc 'Build for Android'
  lane :build do
    gradle(task: 'clean', project_dir: 'android/')
    gradle(task: 'bundle', build_type: 'Release', project_dir: 'android/')
  end

  desc 'Build the Android application and upload to Play store.'
  lane :beta do
    gradle(task: 'clean', project_dir: 'android/')
    increment_version_code(gradle_file_path: "./android/app/build.gradle")
    gradle(task: 'bundle', build_type: 'Release', project_dir: 'android/')
    supply(track: 'internal', aab: './android/app/build/outputs/bundle/release/app-release.aab')
  end

  lane :bump do
    paths = [
      "android/app/build.gradle"
    ]
    git_add(path: paths)
    git_commit(path: paths, message: "Android Version Bump")
    push_to_git_remote()
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("com.website.AppName") # The bundle identifier of your app
apple_id("appleid@gmail.com") # Your Apple email address

itc_team_id("123456789") # App Store Connect Team ID
team_id("AB1C2345DD") # Developer Portal Team ID

json_key_file("play-store-credentials.json")
package_name("com.website.AppName")

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.205.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                 | Version | Update-Status |
| -------------------------------------- | ------- | ------------- |
| fastlane-plugin-increment_version_code | 0.4.3   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                    | Version      |
| -------------------------------------- | ------------ |
| did_you_mean                           | 1.4.0        |
| atomos                                 | 0.1.3        |
| CFPropertyList                         | 3.0.5        |
| colored2                               | 3.1.2        |
| nanaimo                                | 0.3.0        |
| rexml                                  | 3.2.5        |
| xcodeproj                              | 1.21.0       |
| rouge                                  | 2.0.7        |
| xcpretty                               | 0.3.0        |
| terminal-notifier                      | 2.0.0        |
| unicode-display_width                  | 1.8.0        |
| terminal-table                         | 1.8.0        |
| plist                                  | 3.6.0        |
| public_suffix                          | 4.0.7        |
| addressable                            | 2.8.0        |
| multipart-post                         | 2.0.0        |
| word_wrap                              | 1.0.0        |
| optparse                               | 0.1.1        |
| tty-screen                             | 0.8.1        |
| tty-cursor                             | 0.7.1        |
| tty-spinner                            | 0.9.3        |
| artifactory                            | 3.0.15       |
| babosa                                 | 1.0.4        |
| colored                                | 1.2          |
| highline                               | 2.0.3        |
| commander                              | 4.6.0        |
| excon                                  | 0.92.2       |
| faraday-em_http                        | 1.0.0        |
| faraday-em_synchrony                   | 1.0.0        |
| faraday-excon                          | 1.1.0        |
| faraday-httpclient                     | 1.0.1        |
| faraday-multipart                      | 1.0.3        |
| faraday-net_http                       | 1.0.1        |
| faraday-net_http_persistent            | 1.2.0        |
| faraday-patron                         | 1.0.0        |
| faraday-rack                           | 1.0.0        |
| faraday-retry                          | 1.0.3        |
| ruby2_keywords                         | 0.0.5        |
| faraday                                | 1.10.0       |
| unf_ext                                | 0.0.8.1      |
| unf                                    | 0.1.4        |
| domain_name                            | 0.5.20190701 |
| http-cookie                            | 1.0.4        |
| faraday-cookie_jar                     | 0.0.7        |
| faraday_middleware                     | 1.2.0        |
| fastimage                              | 2.2.6        |
| gh_inspector                           | 1.1.3        |
| json                                   | 2.3.0        |
| mini_magick                            | 4.11.0       |
| naturally                              | 2.2.1        |
| rubyzip                                | 2.3.2        |
| security                               | 0.1.3        |
| xcpretty-travis-formatter              | 1.0.1        |
| dotenv                                 | 2.7.6        |
| simctl                                 | 1.6.8        |
| jwt                                    | 2.3.0        |
| uber                                   | 0.1.0        |
| declarative                            | 0.0.20       |
| trailblazer-option                     | 0.1.2        |
| representable                          | 3.1.1        |
| retriable                              | 3.1.2        |
| mini_mime                              | 1.1.2        |
| memoist                                | 0.16.2       |
| multi_json                             | 1.15.0       |
| os                                     | 1.1.4        |
| signet                                 | 0.16.1       |
| googleauth                             | 1.1.2        |
| httpclient                             | 2.8.3        |
| webrick                                | 1.6.1        |
| google-apis-core                       | 0.4.2        |
| google-apis-playcustomapp_v1           | 0.7.0        |
| google-apis-androidpublisher_v3        | 0.18.0       |
| google-cloud-env                       | 1.6.0        |
| google-cloud-errors                    | 1.2.0        |
| google-cloud-core                      | 1.6.0        |
| google-apis-iamcredentials_v1          | 0.10.0       |
| google-apis-storage_v1                 | 0.13.0       |
| rake                                   | 13.0.1       |
| digest-crc                             | 0.6.4        |
| google-cloud-storage                   | 1.36.1       |
| emoji_regex                            | 3.2.3        |
| jmespath                               | 1.6.1        |
| aws-partitions                         | 1.575.0      |
| aws-eventstream                        | 1.2.0        |
| aws-sigv4                              | 1.4.0        |
| aws-sdk-core                           | 3.130.0      |
| aws-sdk-kms                            | 1.55.0       |
| aws-sdk-s3                             | 1.113.0      |
| uri                                    | 0.10.0       |
| claide                                 | 1.1.0        |
| bundler                                | 2.3.11       |
| forwardable                            | 1.3.1        |
| logger                                 | 1.4.2        |
| cgi                                    | 0.1.0.1      |
| date                                   | 3.0.3        |
| timeout                                | 0.1.0        |
| stringio                               | 0.1.0        |
| ipaddr                                 | 1.2.2        |
| openssl                                | 2.1.3        |
| zlib                                   | 1.1.0        |
| mutex_m                                | 0.1.0        |
| ostruct                                | 0.2.0        |
| strscan                                | 1.0.3        |
| io-console                             | 0.5.6        |
| delegate                               | 0.1.0        |
| fileutils                              | 1.4.1        |
| singleton                              | 0.1.0        |
| open3                                  | 0.1.0        |
| yaml                                   | 0.1.0        |
| psych                                  | 3.1.0        |
| etc                                    | 1.1.0        |
| fastlane-plugin-increment_version_code | 0.4.3        |
</details>

*generated on:* **2022-05-16**
</details>
WARN [2022-05-16 02:23:18.13]: Take notice that this output may contain sensitive information, or simply information that you don't want to make public.
INFO [2022-05-16 02:23:18.13]: Open https://github.com/fastlane/fastlane/issues/new to submit a new issue ✅
CircleCI received exit code 0 </pre>
</details>

**Comment by lucgrabowski** [comment]
This doesn't seem to be a fastlane issue. Have you created a build locally, uploaded it to Testflight and checked if you get the result?

**Comment by umrashrf** [comment]
Using the development certificate and provisioning profile generated by _fastlane match_ gives me an error when running the app on my iPhone. The build was successful but app install was failed.

Error: **Unable to install app AppName**

More Logs:

<Details> 
<pre> Details
Unable to install "AppName"
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620395
User Info: {
    DVTErrorCreationDateKey = "2022-05-16 23:42:48 +0000";
    IDERunOperationFailingWorker = IDEInstalliPhoneLauncher;
}
--
A valid provisioning profile for this executable was not found.
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620395
User Info: {
    DVTRadarComponentKey = 487925;
    MobileDeviceErrorCode = "(0xE8008015)";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x000000011fd19614 DTDKCreateNSErrorFromAMDErrorCode + 272
	1   DTDeviceKitBase                     0x000000011fd52dd8 __90-[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:]_block_invoke + 160
	2   DVTFoundation                       0x0000000105c4de58 DVTInvokeWithStrongOwnership + 76
	3   DTDeviceKitBase                     0x000000011fd52b30 -[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:] + 1336
	4   IDEiOSSupportCore                   0x000000013b451590 __118-[DVTiOSDevice(DVTiPhoneApplicationInstallation) processAppInstallSet:appUninstallSet:installOptions:completionBlock:]_block_invoke.301 + 2916
	5   DVTFoundation                       0x0000000105d75194 __DVT_CALLING_CLIENT_BLOCK__ + 16
	6   DVTFoundation                       0x0000000105d762ac __DVTDispatchAsync_block_invoke + 364
	7   libdispatch.dylib                   0x00000001c00ec604 _dispatch_call_block_and_release + 32
	8   libdispatch.dylib                   0x00000001c00ee1c8 _dispatch_client_callout + 20
	9   libdispatch.dylib                   0x00000001c00f58a8 _dispatch_lane_serial_drain + 668
	10  libdispatch.dylib                   0x00000001c00f6404 _dispatch_lane_invoke + 392
	11  libdispatch.dylib                   0x00000001c0100c98 _dispatch_workloop_worker_thread + 648
	12  libsystem_pthread.dylib             0x00000001c02ae360 _pthread_wqthread + 288
	13  libsystem_pthread.dylib             0x00000001c02ad080 start_wqthread + 8
);
}
--

Analytics Event: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_model" = "iPhone14,3";
    "device_osBuild" = "15.4.1 (19E258)";
    "device_platform" = "com.apple.platform.iphoneos";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = arm64;
    "operation_duration_ms" = 2884;
    "operation_errorCode" = "-402620395";
    "operation_errorDomain" = "com.apple.dt.MobileDeviceErrorDomain";
    "operation_errorWorker" = IDEInstalliPhoneLauncher;
    "operation_name" = IDEiPhoneRunOperationWorkerGroup;
    "param_consoleMode" = 0;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 5;
    "param_destination_isProxy" = 0;
    "param_destination_platform" = "com.apple.platform.iphoneos";
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 0;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 0;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 0;
    "param_launcher_substyle" = 0;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_runnable_swiftVersion" = "5.6";
    "param_runnable_type" = 2;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphoneos15.4";
    "sdk_osVersion" = "15.4";
    "sdk_variant" = iphoneos;
}
--


System Information

macOS Version 12.3.1 (Build 21E258)
Xcode 13.3.1 (20103) (Build 13E500a)
Timestamp: 2022-05-16T19:42:48-04:00 </pre> </Details>

**Comment by lucgrabowski** [comment]
The problem isn't related to fastlane so I've moved it to discussions so that the community could help you. Here are some hints from me ;)
Go to project in Xcode, select Signing and capabilities tab and set this profile to see what is the problem. Maybe the development device that you try to use is not available in this development profile.
https://developer.apple.com/documentation/xcode/distributing-your-app-to-registered-devices
Also, check code signing identity in project settings.

**Reply by umrashrf** [reply]
I can confirm that there is no error in XCode. The Xcode build is successful. I did register my device with fastlane match and appropriate provisioning profile was created. Do I need to add the provisioning profile generated by fastlane match to my device somehow?

**Reply by lucgrabowski** [reply]
Check your profile on Apple Developer Center if your device is there on a list of devices and later if it is in provisioning profile. If the profile is wrong, regenerate it with match.

**Reply by umrashrf** [reply]
You could be right. I recently changed my iPhone so the UDID is of an older phone. I will verify that and fix that if that's the case.

**Reply by umrashrf** [reply]
I have verified that my device id is there on developer.apple.com. After building in Xcode, Xcode tries to install app on the device and fails with this error "A valid provisioning profile for this executable was not found.".

I can also confirm that this is not the problem if I use Xcode <img width="295" alt="image" src="https://user-images.githubusercontent.com/652326/168869499-4abe398e-9041-462a-a521-3685dacb685f.png">

**Reply by umrashrf** [reply]
Ok I think I found the problem.

Changed:
Edit Schemes > Run > Build Configuration > Release 
To:
Edit Schemes > Run > Build Configuration > Debug

**Comment by umrashrf** [comment]
The app works on my iPhone with Fastlane Match Development certificates and profiles but does not work if published to TestFlight and downloaded from TestFlight. This is a major blocker in adapting Fastlane so I will go back to using Xcode automatic profile provisioning for the moment.
